ES6JSX branch gives no-multi-str errors on JSX

I'm using eslint v0.11.0-alpha.0 and getting `no-mult-str` errors in my JSX template code:

```JavaScript
var React = require('react');

var App = React.createClass({
  render: function() {
    return (
      <div>
        <h1>wat</h1>
      </div>
    );
  }
});

React.render(<App />, document.body);
```

This produces the following errors:

```bash
ಠ_ಠ  ~/code/happytrack λ eslint durp.jsx

durp.jsx
  3:4   error  App is defined but never used                                 no-unused-vars
  6:11  error  Multiline support is limited to browsers supporting ES5 only  no-multi-str
  7:20  error  Multiline support is limited to browsers supporting ES5 only  no-multi-str

✖ 3 problems
```

The first error looks like it is from https://github.com/estools/escope/issues/33, but the last two errors seem incorrect. I expected the es6jsx branch to correctly parse JSX without giving no-multi-str errors. Am I doing something wrong?
<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/6618490-es6jsx-branch-gives-no-multi-str-errors-on-jsx?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/lib/rules/no-multi-str.js
+++ b/lib/rules/no-multi-str.js
@@ -1,6 +1,8 @@
 /**
  * @fileoverview Rule to flag when using multiline strings
  * @author Ilya Volodin
+ * @copyright 2014 Nicholas C. Zakas. All rights reserved.
+ * @copyright 2013 Ilya Volodin. All rights reserved.
  */
 
 //------------------------------------------------------------------------------
@@ -10,11 +12,26 @@
 module.exports = function(context) {
     "use strict";
 
+    /**
+     * Determines if a given node is part of JSX syntax.
+     * @param {ASTNode} node The node to check.
+     * @returns {boolean} True if the node is a JSX node, false if not.
+     * @private
+     */
+    function isJSXElement(node) {
+        return node.type.indexOf("JSX") === 0;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public API
+    //--------------------------------------------------------------------------
+
     return {
 
         "Literal": function(node) {
             var lineBreak = /\n/;
-            if (lineBreak.test(node.raw)) {
+
+            if (lineBreak.test(node.raw) && !isJSXElement(node.parent)) {
                 context.report(node, "Multiline support is limited to browsers supporting ES5 only.");
             }
         }

--- a/tests/lib/rules/no-multi-str.js
+++ b/tests/lib/rules/no-multi-str.js
@@ -17,7 +17,8 @@ var eslint = require("../../../lib/eslint"),
 var eslintTester = new ESLintTester(eslint);
 eslintTester.addRuleTest("lib/rules/no-multi-str", {
     valid: [
-        "var a = 'Line 1 Line 2';"
+        "var a = 'Line 1 Line 2';",
+        { code: "var a = <div>\n<h1>Wat</h1>\n</div>;", ecmaFeatures: { jsx: true }}
     ],
     invalid: [
         { code: "var x = 'Line 1 \\\n Line 2'", errors: [{ message: "Multiline support is limited to browsers supporting ES5 only.", type: "Literal"}] },

